modules/tinymce/src/core/main/ts/textpatterns/core/BlockPattern.ts

```diff
@@ -23,6 +26,20 @@ const stripPattern = (dom: DOMUtils, block: Node, pattern: BlockPattern): void =
 
       Utils.deleteRng(dom, rng, (e: Node) => e === block);
     });
+
+    /**
+     * TINY-9603: If there is a single space between pattern.start and text (e.g. # 1)
+     * then it will be left in the text content and then can appear in certain circumstances.
+     * This is not an issue with multiple spaces because they are transformed to non-breaking ones.
+     *
+     * In this specific case we've decided to remove this single space whatsoever
+     * as it feels to be the expected behavior.
+     */
+    const text = SugarElement.fromDom(node);
+    const textContent = SugarText.get(text);
+    if (startsWithSingleSpace(textContent)) {
+      SugarText.set(text, textContent.replace(/^\s/, ''));
```

Comment:
Do we need regex for this, or can we just exclude the first character? `textContent.slice(1)` for example?

Reply:
Yeah, we can simplify it with `slice(`)`. Thanks! Updated.